slrealizer/realize_sl.py

```diff
@@ -0,0 +1,313 @@
+    using the OM10 catalog and observation history provided in constructor.
+    '''
+
+    def __init__(self, observation):
+        """
+        Reads in a lens sample catalog and observation data.
+        We assume lenses are OM10 lenses and observation file is a pandas df
+        """
+        # TODO query outside class and make resulting matrix a parameter
+        self.observation = observation
+        self.num_obs = len(self.observation)
+        
+        # GalSim drawImage params
+        self.fft_params = galsim.GSParams(maximum_fft_size=10240)
+        self.pixel_scale = 0.1
+        self.nx, self.ny = 48, 48 
+        
+        # Source table column list
+        self.sourceCols = ['MJD', 'filter', 'x', 'y', 'flux', 'trace', 'skyErr', 'e1', 'e2', 'psf_fwhm', 'objectId']
+        
+    def get_observation(self, obsID=None, rownum=None):
+        if obsID is not None and rownum is not None:
+            raise ValueError("Need to define either obsID or rownum, not both.")
+        
+        if obsID is not None:
+            return self.observation.loc[self.observation['obsHistID']==obsID]
+        elif rownum is not None:
+            return self.observation.loc[rownum]
+    
+    def _get_lens(self, lensID):
+        # This function will depend on the format of each lens catalog
+        raise NotImplementedError
+        
+    def draw_system(self, galsimInput, obsInfo, save_dir=None):
+        '''
+        Draws all objects of the given lens system
+        in the given observation conditions using GalSim
+
+        Keyword arguments:
+        lensInfo -- a row of the OM10 DB
+        obsInfo -- a row of the observation history df
+        save_dir -- directory in which to save the image
+
+        Returns:
+        A GalSim object of the aggregate system used to render
+        the image
+        '''
+        histID, MJD, band, PSF_FWHM, sky_mag = obsInfo
+
+        # Lens galaxy
+        galaxy = galsim.Gaussian(half_light_radius=galsimInput['half_light_radius'],\
+                                 flux=galsimInput['flux'])\
+                       .shear(e=galsimInput['e'], beta=galsimInput['beta'])
+        # Lensed quasar
+        for i in xrange(galsimInput['num_objects']):
+            lens = galsim.Gaussian(flux=galsimInput['flux_'+str(i)], sigma=0.0)\
+                         .shift(galsimInput['xy_'+str(i)])
+            galaxy += lens
+            
+        psf = galsim.Gaussian(flux=1.0, fwhm=PSF_FWHM)
+        galsim_obj = galsim.Convolve([galaxy, psf], gsparams=self.fft_params)
+        galsim_img = galsim_obj.drawImage(nx=self.nx, ny=self.ny, scale=self.pixel_scale)
+        if save_dir is not None:
+            plt.imshow(galsim_img.array, interpolation='none', aspect='auto')
+            plt.savefig(save_dir+'before_deblend.png')
+            plt.close()
+        return galsim_img
+        
+    def estimate_hsm(self, galsim_img, obsInfo):
+        """
+        Performs HSM shape estimation on the image 
```

Comment:
Looks like this method always uses GalSim to estimate the moments-  the docstring should say this, I think.

Reply:
Yes! HSM is a shape estimation method (`FindAdaptiveMom`) of GalSim's Image class. Updated docstring to be clearer.